File: java/src/com/alofmethbin/jadawin/styles/DatedEvents.java
package com.alofmethbin.jadawin.styles;

import com.alofmethbin.jadawin.Page;
import java.time.LocalDate;

public class DatedEvents extends Events {

    @Override
    public String indexTitle(Page page) {
        LocalDate d = page.getDate();
        if (d != null) {
            return super.indexTitle( page) + 
                   " (" +
                   d.getMonth().toString().substring(0, 3) +
                   " " +
                   d.getYear() +
                   ")"; 
        } else {
            page.error( "Missing date for DatedEvents page");
            return super.indexTitle( page);
        }
    }
}

File: ruby/styles/dated_events.rb
require_relative 'events'

module Styles
  class DatedEvents < Events
    def index_title( page)
      if d = page.date
        super( page) + page.date.strftime( ' (%b %Y)')
      else
        page.error( 'Missing date for DatedEvents page')
        super( page)
      end
    end
  end
end
